### PR TITLE
Fix Datafusion system tests

### PR DESCRIPTION
**Pull request by VladaZakharova**
- Fix naming in system tests
- Add handling of 409 exception for requests. This error could occur in cases when previous started request on Datafusion pipeline was not finished yet and the next request started. 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/system/providers/google/cloud/datafusion/example_datafusion_async.py [hunk 1]
```diff
@@ -145,19 +146,19 @@
 # [END howto_data_fusion_env_variables]
 
 with models.DAG(
-    "example_data_fusion_async",
+    DAG_ID,
     start_date=datetime(2021, 1, 1),
     catchup=False,
```
Comment (moiseenkov) [inline tests/system/providers/google/cloud/datafusion/example_datafusion_async.py:151]: Could you please add tags into both dags?

ref: https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-47+New+design+of+Airflow+System+Tests
"Process of migration in details" -> section 5.b

File: tests/system/providers/google/cloud/datafusion/example_datafusion_async.py [hunk 2]
```diff
@@ -233,40 +234,39 @@
...
+        task_id="delete_bucket1", bucket_name=BUCKET_NAME_1, trigger_rule=TriggerRule.ALL_DONE
+    )
+    delete_bucket2 = GCSDeleteBucketOperator(
+        task_id="delete_bucket2", bucket_name=BUCKET_NAME_1, trigger_rule=TriggerRule.ALL_DONE
+    )
 
     (
-        create_bucket1
-        >> create_bucket2
+        [create_bucket1, create_bucket2]
```
Comment (moiseenkov) [inline tests/system/providers/google/cloud/datafusion/example_datafusion_async.py:258]: Can we add comments (in both tests) that would separate preparation, test, and tear down section?
https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-47+New+design+of+Airflow+System+Tests

Something like
```python
(
    # TEST SETUP
    create_dataset
    >> create_table
    # TEST BODY
    >> copy_selected_data
    # TEST TEARDOWN
    >> delete_dataset
)
```

File: tests/system/providers/google/cloud/datafusion/example_datafusion.py [hunk 3]
```diff
@@ -36,34 +35,36 @@
     CloudDataFusionStopPipelineOperator,
     CloudDataFusionUpdateInstanceOperator,
 )
-from airflow.providers.google.cloud.operators.gcs import GCSCreateBucketOperator
+from airflow.providers.google.cloud.operators.gcs import GCSCreateBucketOperator, GCSDeleteBucketOperator
```
Comment (moiseenkov) [inline tests/system/providers/google/cloud/datafusion/example_datafusion.py:38]: From my perspective that would look better, but it's up to you. Both variants are good for me.
```python
from airflow.providers.google.cloud.operators.gcs import (
    GCSCreateBucketOperator,
    GCSDeleteBucketOperator,
)
```

**Review by MaksYermak — APPROVED** [review]

File: tests/system/providers/google/cloud/datafusion/example_datafusion_async.py (hunk 1 above)
Comment (VladaZakharova) [reply tests/system/providers/google/cloud/datafusion/example_datafusion_async.py:151]: Sure, thank you!

**Review by moiseenkov — APPROVED** [review]

**Pull request closed without merging**